timeboost-sequencer/src/lib.rs

```diff
@@ -350,11 +356,15 @@ impl Task {
         let mut candidates = Vec::new();
         while !actions.is_empty() {
             let mut round = RoundNumber::genesis();
+            let mut evidence = Evidence::Genesis;
             let mut lists = Vec::new();
             while let Some(action) = actions.pop_front() {
                 match action {
                     Action::Deliver(payload) => {
                         round = payload.round();
+                        if payload.evidence().round() > evidence.round() {
+                            evidence = payload.evidence().clone()
+                        }
```

Comment:
Maybe it makes sense to move this into the `Ok(data)` branch below so the values are only updated if the corresponding candidate list has been successfully decoded.

Once done, using a hypothetical `Payload::into_evidence` method would save a clone.